sound/soc/sof/intel/hda-ctrl.c

```diff
@@ -148,6 +149,24 @@ void hda_dsp_ctrl_misc_clock_gating(struct snd_sof_dev *sdev, bool enable)
 	snd_sof_pci_update_bits(sdev, PCI_CGCTL, PCI_CGCTL_MISCBDCGE_MASK, val);
 }
 
+void hda_dsp_ctrl_clock_power_gating(struct snd_sof_dev *sdev, bool enable)
+{
+	struct hdac_bus *bus = sof_to_bus(sdev);
+	u32 val;
+
+	/* Update PDCGE bit of CGCTL register */
+	val = enable ? PCI_CGCTL_ADSPDCGE : 0;
```

Comment:
I think this is opposite for PCI_TCSEL_ADSPPGD?  

Be careful that some bits are gating disable(e.g. ADSPPGD), some others are gating enable(e.g. PCI_CGCTL_ADSPDCGE), we need to handle them carefully.

Reply:
@keyonjie yes, but the code does the right thing here. When we want to enable clock gating we write 1 to PCI_CGCTL_ADSPDCGE in CGCTL and write 0 to PCI_PGCTL BIT(2).